apps/api-extractor/src/generators/ApiReportGenerator.ts

```diff
@@ -93,6 +94,13 @@ export class ApiReportGenerator {
     // Emit the regular declarations
     for (const entity of collector.entities) {
       const astEntity: AstEntity = entity.astEntity;
+      const symbolMetadata: SymbolMetadata | undefined = collector.tryFetchMetadataForAstEntity(astEntity);
+      const maxEffectiveReleaseTag: ReleaseTag = symbolMetadata?.maxEffectiveReleaseTag ?? ReleaseTag.None;
+
+      if (!this._shouldIncludeReleaseTag(maxEffectiveReleaseTag, reportVariant)) {
```

Comment:
This is the crux of the fix. We need to discard the entire entity if it doesn't meet the threshold for the API report. This is what the `d.ts` trimming generator does.